packages/kit/src/core/prerender/prerender.js

```diff
@@ -14,25 +14,17 @@ import { escape_html_attr } from '../../utils/escape.js';
  * @typedef {import('types').Logger} Logger
  */
 
-/** @type {(details: Parameters<PrerenderErrorHandler>[0] ) => string} */
-function format_error({ status, path, referrer, referenceType }) {
-	return `${status} ${path}${referrer ? ` (${referenceType} from ${referrer})` : ''}`;
-}
+/**
+ * @param {Parameters<PrerenderErrorHandler>[0]} details
+ * @param {import('types').ValidatedKitConfig} config
+ */
+function format_error({ status, path, referrer, referenceType }, config) {
+	const message =
+		status === 404 && !path.startsWith(config.paths.base)
+			? `${path} does not begin with \`base\`, which can be imported from \`$app/paths\``
```

Comment:
Is this even more clear or too verbose already? I err on the side of verbosity, in bigger projects, someone new could be like "what's this base thing, and where does it come from"?

```suggestion
			? `${path} does not begin with \`base\`, which is configured in \`paths.base\` and can be imported from \`$app/paths\``
```

Reply:
yeah, i like that